project_euler/problem_107/sol1.c

```diff
@@ -0,0 +1,188 @@
+/**
+ * \file
+ * \brief Project Euler problem 107 (https://projecteuler.net/problem=107)
+ * solution using Kruskals
+ * \author [David Santamaria](https://github.com/Daves1245)
+ */
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <assert.h>
+
+#define NUM_VERTICES 40
+#define NUM_LINES 40
+#define MAX_LINE_LEN 4 * 40 // a line is at most 40 3-digit numbers and a comma
+
+/*
+ * We represent an edge as a 3 tuple (u, v, w),
+ * where u and v are the vetices connected by
+ * the edge, and w is the weight of the edge
+ */
+struct edge {
+  int u;
+  int v;
+  int w;
+};
+
+/*
+ * The union find data structure
+ */
+struct ufds {
+  int rank[NUM_VERTICES], parent[NUM_VERTICES], size[NUM_VERTICES];
```

Comment:
```suggestion
  int rank[NUM_VERTICES]; /**< 1-line description of variable here */
  int parent[NUM_VERTICES];/**< 1-line description of variable here */
  int size[NUM_VERTICES];/**< 1-line description of variable here */
```

Reply:
Fixed in 67e24f42cf2c3a11b780ff3b2af496002cc2ae71